include/exadg/utilities/throughput_parameters.h

```diff
@@ -123,10 +124,11 @@ struct ThroughputParameters
   void
   print_results(MPI_Comm const & mpi_comm)
   {
-    print_throughput(wall_times, operator_type, mpi_comm);
+    print_throughput(wall_times, ExaDG::Utilities::enum_to_string(operator_type), mpi_comm);
   }
 
-  std::string operator_type = "Undefined";
+  // type of PDE operator
+  OperatorType operator_type = OperatorType();
```

Comment:
The most essential change of this PR, which causes to template `ThroughputParameters`. The rest is technical.

Reply:
Does this work? I think we should use `OperatorType operator_type=OperatorType::Undefined` instead. This requires the `OperatorType` to have a field `Undefined` and therefore the other approach would be more generic, but I think it does not work. If I test the current initialization on a minimum example, it seems the initialization does not point to a valid enum:
```
enum class Test
{
  a = 1,
  b
};

int
main()
{
  Test t = Test();
  std::cout << "enum name " << magic_enum::enum_name(t) << std::endl;
  return 0;
}
Output: enum name
```

Reply:
I have found this https://lifecs.likai.org/2010/07/c-enum-default-constructor.html, which is quite similar to your example.

But doesn't the new enum functionality check anyway if the string provided in the input file can be translated to one of the values of the enum type? I thought we would receive an error message then? Or do you mean if we do not set `true` as last parameter in `add_parameter(..., true)`?

Reply:
Can you also try with a non-existing value for `OperatorType`? Maybe an empty `Selection` pattern allows for all values, and we don't get the nice error messages from the `ParameterHandler` but instead error messages that stings could not be converted to the enum. 

And I think `OperatorType operator_type = OperatorType();` is equivalent to `OperatorType operator_type;`. Can you try this? Personally, I am confused by `EnumType = EnumType()` because I don't know what this implies.

Reply:
> And I think OperatorType operator_type = OperatorType(); is equivalent to OperatorType operator_type;. Can you try this? Personally, I am confused by EnumType = EnumType() because I don't know what this implies.

No, this is not the same. This is what I reported a few days ago and what I observed again today. Which is why I added `= OperatorType()`.

Reply:
In my minimal example, it is the same. Since also the issue you reported works on my machine, can we just go with `OperatorType operator_type=OperatorType::Undefined` since we know exactly what we are doing in this case? 

Alternatively, we could introduce another enum utility function to initialize with the first value
```
template <typename EnumType>
EnumType
first_value()
{
return magic_enum::enum_values<EnumType>()[0];
}
```